packages/platform-bun/package.json

```diff
@@ -49,7 +49,7 @@
   },
   "devDependencies": {
     "@effect/platform": "workspace:^",
-    "bun-types": "1.1.22",
+    "bun-types": "1.1.36",
```

Comment:
It's [recommended](https://bun.sh/guides/runtime/typescript) to use the `@types/bun` shim since [1.0.19](https://bun.sh/blog/bun-v1.0.19#re-introducing-types-bun-formerly-bun-types). Also, any reason not to use a caret constraint?